### PR TITLE
Add schedules to new model

### PR DESCRIPTION
**Pull request by peteryates**
This is a cut down, simplified version of what's in ECF. The Schedule is a join table between course and cohort, it has a start date, a declaration_type (enum, for now), name.

image

### CODE REVIEW

**Comment by peteryates** [comment]
The ids are `int` for now, I think we should probably stick with them and just use `uuds` on things visible via the API.

**Comment by peteryates** [comment]
Thoughts:

* unique index on `course_id`/`cohort_id`/`starts_on`
* do we _really need_ the name?
* use `declaration_type` in the declarations table too

File: data-modelling.md [hunk 1]
```diff
@@ -157,6 +160,15 @@ erDiagram
         uuid declaration_id
         string state
     }
+
+    Schedule {
```
Comment (cwrw) [inline data-modelling.md:164]: maybe we can also add `schedule_id` to the application?

**Comment by cwrw** [comment]
I think we need the many to many between a course and a schedule,  a course can have many schedules in a cohort, and a schedule can belong to many courses in a cohort

**Comment by peteryates** [comment]
Is this similar to the `course_group` idea?

```mermaid
erDiagram
  CourseGroup ||--|{ Course : ""
  Cohort ||--|{ Schedule : ""
  CourseGroup ||--|{ Schedule : ""  

```

File: data-modelling.md [hunk 2]
```diff
@@ -71,14 +76,17 @@ erDiagram
 
     Course {
         uuid id
+        uuid course_group_id
         string identifier "Is this unique?"
```
Comment (cwrw) [inline data-modelling.md:80]: remove question

File: data-modelling.md [hunk 3]
```diff
@@ -157,6 +165,22 @@ erDiagram
...
+    }
+
+    Schedule {
+        uuid id
+        uuid course_group_id
+        uuid cohort_id
+        string name
+        date declaration_starts_on
+        date schedule_applies_from
+        date scheudle_applies_to
```
Comment (cwrw) [inline data-modelling.md:181]: fix spelling

File: data-modelling.md [hunk 4]
```diff
@@ -71,14 +76,17 @@ erDiagram
...
         string identifier "Is this unique?"
+
     }
 
     Application {
         uuid id
         uuid course_id
         uuid lead_provider_id
         uuid user_id
+        uuid schedule_id
```
Comment (cwrw) [inline data-modelling.md:89]: add `submitted_at` remove `cohort_id`

**Pull request merged**